### PR TITLE
Add outline to unit count and damage counts

### PR DESCRIPTION
**Pull request by ron-murhammer**
Addresses: https://forums.triplea-game.org/topic/917/inception-a-dream-within-a-dream-unit-stack-numbers-display

**Before:**
image

**After:**
image

**Font Properties:**
image

### CODE REVIEW

File: game-core/src/main/java/games/strategy/triplea/ui/screen/UnitsDrawer.java [hunk 1]
```diff
@@ -197,13 +181,37 @@ private void drawUnit(final Graphics2D graphics, final Image image, final Rectan
...
+    if (territoryName.length() != 0 && bombingUnitDamage > 0) {
+      final String s = String.valueOf(bombingUnitDamage);
+      final int x = placementPoint.x - bounds.x + uiContext.getUnitImageFactory().getUnitImageWidth() / 4;
+      final int y = placementPoint.y - bounds.y + uiContext.getUnitImageFactory().getUnitImageHeight() / 4;
+      drawOutlinedText(graphics, s, x, y, MapImage.getPropertyMapFont(), MapImage.getPropertyUnitFactoryDamageColor(),
+          MapImage.getPropertyUnitFactoryDamageOutline());
+    }
+  }
+
+  private static void drawOutlinedText(final Graphics2D graphics, final String s, final int x, final int y,
```
Comment (ron-murhammer) [inline game-core/src/main/java/games/strategy/triplea/ui/screen/UnitsDrawer.java]: This method is the core change. Everything else is pretty boiler plate.

**Comment by Heppisorus** [comment]
Looks good Red.  Well done as usual.

**Comment by ron-murhammer** [comment]
@RoiEXLab Yeah, a lot of it is setter/getter/reset properties boiler plate code. Probably could create some sort of utility to try to simplify them a bit but didn't really spend time digging into them just followed the existing pattern as the main change updating how counts are draw and the properties were kind of just a customization add-on for outlines.

File: game-core/src/main/java/games/strategy/triplea/image/MapImage.java [hunk 2]
```diff
@@ -96,46 +143,82 @@ public static void setPropertyUnitCountColor(final Color color) {
...
   public static void resetPropertyUnitHitDamageColor() {
     final Preferences pref = Preferences.userNodeForPackage(MapImage.class);
     pref.remove(PROPERTY_UNIT_HIT_DAMAGE_COLOR_STRING);
-    propertyUnitHitDamageColor = Color.black;
+    propertyUnitHitDamageColor = UNIT_HIT_DAMAGE_COLOR_DEFAULT;
+  }
+
+  public static void resetPropertyUnitHitDamageOutline() {
+    final Preferences pref = Preferences.userNodeForPackage(MapImage.class);
+    pref.remove(PROPERTY_UNIT_HIT_DAMAGE_OUTLINE_STRING);
```
[the remark this answers is not in the document]
Comment (ron-murhammer) [reply game-core/src/main/java/games/strategy/triplea/image/MapImage.java]: Yeah, added a reset helper method.

File: game-core/src/main/java/games/strategy/triplea/image/MapImage.java [hunk 3]
```diff
@@ -47,37 +64,67 @@ public static Color getPropertyTerritoryNameAndPuAndCommentColor() {
...
+
   public static Color getPropertyUnitHitDamageColor() {
     if (propertyUnitHitDamageColor == null) {
       final Preferences pref = Preferences.userNodeForPackage(MapImage.class);
       propertyUnitHitDamageColor =
-          new Color(pref.getInt(PROPERTY_UNIT_HIT_DAMAGE_COLOR_STRING, Color.black.getRGB()));
+          new Color(pref.getInt(PROPERTY_UNIT_HIT_DAMAGE_COLOR_STRING, UNIT_HIT_DAMAGE_COLOR_DEFAULT.getRGB()));
     }
     return propertyUnitHitDamageColor;
   }
```
[the remark this answers is not in the document]
Comment (ron-murhammer) [reply game-core/src/main/java/games/strategy/triplea/image/MapImage.java]: Not really sure much can be done for get/set as the variation is too high without doing significant refactoring or only covering a portion of them.

**Pull request merged**